src/jit/emitxarch.cpp

```diff
@@ -95,7 +95,8 @@ bool emitter::IsDstDstSrcAVXInstruction(instruction ins)
             ins == INS_pmaxuw || ins == INS_pmaxud || ins == INS_vinserti128 || ins == INS_punpckhbw ||
             ins == INS_punpcklbw || ins == INS_punpckhqdq || ins == INS_punpcklqdq || ins == INS_punpckhwd ||
             ins == INS_punpcklwd || ins == INS_punpckhdq || ins == INS_packssdw || ins == INS_packsswb ||
-            ins == INS_packuswb || ins == INS_packusdw || ins == INS_vperm2i128);
+            ins == INS_packuswb || ins == INS_packusdw || ins == INS_vperm2i128 || ins == INS_roundps ||
+            ins == INS_roundss || ins == INS_roundpd || ins == INS_roundsd);
```

Comment:
- `roundss` and `roundsd` should be `DstSrcSrc` instructions to avoid the similar register dependency issue to #14225.
- `roundps` and `roundpd` are two-register instructions in AVX, so you should remove here.

Reply:
> roundps and roundpd are two-register instructions in AVX, so you should remove here.

What do you mean? The manual indicates they are 3 operand `RMI` instructions:
* ROUNDPD xmm1, xmm2/m128, imm8
* VROUNDPD xmm1, xmm2/m128, imm8
* VROUNDPD ymm1, ymm2/m256, imm8
*
* ROUNDPS xmm1, xmm2/m128, imm8
* VROUNDPS xmm1, xmm2/m128, imm8
* VROUNDPS ymm1, ymm2/m256, imm8

This is slightly different from teh scalar versions which are 3 operand `RMI` for SSE41 and 4 operand `RVMI` for AVX:
* ROUNDSD xmm1, xmm2/m64, imm8
* VROUNDSD xmm1, xmm2, xmm3/m64, imm8
*
* ROUNDSS xmm1, xmm2/m32, imm8
* VROUNDSS xmm1, xmm2, xmm3/m32, imm8

> roundss and roundsd should be DstSrcSrc instructions to avoid the similar register dependency issue to #14225.

Isn't `DPPS` and `DPPD` also in that category? They are `RIM` instructions: `DPPS xmm1, xmm2/m128, imm8` and `DPPD xmm1, xmm2/m128, imm8`.

It seems like there are some inefficiencies in the INST table in general, such as it not having entries for `RMI` or `RVMI` encoded instructions. This leads to these instructions being listed as `RM` encoded, and having special checks/handling elsewhere (as per the comment on this method)

Reply:
The two functions `IsDstDstSrcAVXInstruction` and `IsDstSrcSrcAVXInstruction` are only used to automatically convert SSE instructions to the corresponding VEX.NDS instructions that use `VEX.vvvv` to encode the additional register. For example, `roundss xmm0, xmm1, imm8` -> `vroundss xmm0, xmm1, xmm2, imm8` where xmm1 in VEX.NDS version is encoded by `VEX.vvvv`. But `roundps/d` has the same registers as its VEX.NDS version.

> Isn't DPPS and DPPD also in that category? They are RIM instructions: DPPS xmm1, xmm2/m128, imm8 and DPPD xmm1, xmm2/m128, imm8.

No, this issue is just from scalar instructions because certain scalar floating-point instructions need an additional register to specify the upper section of xmm.
> Round the low packed single precision floating-point value in xmm3/m32 and place the result in xmm1. The rounding mode is determined by imm8. Also, **upper packed single precision floating-point values (bits[127:32]) from xmm2 are copied to xmm1[127:32].**

> It seems like there are some inefficiencies in the INST table in general

Yes, so we are considering to refactor it #14523

Reply:
Fixed.